coderd/provisionerdserver/provisionerdserver_test.go

```diff
@@ -1569,31 +1569,42 @@ func TestInsertWorkspaceResource(t *testing.T) {
 func TestNotifications(t *testing.T) {
 	t.Parallel()
 
-	t.Run("Workspace deletion", func(t *testing.T) {
+	t.Run("Workspace Events", func(t *testing.T) {
```

Comment:
I'd prefer if we made a new test for the autobuild cases; it's complicating the workspace deletion tests, and we'll doubtless need to add further tests for workspace events which would further add complexity if we followed this pattern.

Also: it doesn't seem like you're explicitly testing this case:

```go
if build.Reason.Valid() && build.Reason == database.BuildReasonInitiator {
	return // failed workspace builds initiated by a user should not notify
}
```

Reply:
Fixed